From 99a4f7812e6a1d2b8ff57c82c0ac32e8de1834bf Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Thu, 10 Jul 2014 12:39:20 +0800
Subject: [PATCH 1783/1825] fix: alp: pon: correct EPON init configuration for
 Serdes.

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit a2df3b93e11b4b8cbc2ab0f77a35e82de7aabf42

EPON/P2P:
after writing 0xfc00 to 0xf1032004:
Set value 0x14 to reg. 0x4D (0xf1032134);
Remove writing: asicOntMiscRegWrite(mvAsicReg_PON_SERDES_INTERNAL_REG_51, 0x104;
Set the correct amplitude: value 0xc94c to reg.0xd(0xf1032034)

Signed-off-by: Victor Gu <xigu@marvell.com>

Change-Id: Ib5af75b63182f797b3b5b67df6b7027800892761
Reviewed-on: http://vgitil04.il.marvell.com:8080/9112
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
Tested-by: Yehuda Yitschak <yehuday@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../mv_drivers_lsp/mv_pon/plat/epon/eponOnuBoard.c |   17 ++++++++++++-----
 1 files changed, 12 insertions(+), 5 deletions(-)

diff --git a/arch/arm/plat-armada/mv_drivers_lsp/mv_pon/plat/epon/eponOnuBoard.c b/arch/arm/plat-armada/mv_drivers_lsp/mv_pon/plat/epon/eponOnuBoard.c
index cc6e220..481c50c 100644
--- a/arch/arm/plat-armada/mv_drivers_lsp/mv_pon/plat/epon/eponOnuBoard.c
+++ b/arch/arm/plat-armada/mv_drivers_lsp/mv_pon/plat/epon/eponOnuBoard.c
@@ -429,17 +429,24 @@ MV_STATUS onuEponSerdesInit(void)
 
 	/* Internal Regs */
 	/* ============= */
-	/* Set power_reg0	    (0xd0032004 set 0xfc00);recheck bits 7,2,3 */
-	/* Set register 0x51    (0xd0032144 set 0x104); interface_reg2, Interface and Digital Reference Clock Control */
-	/* Set register 0x3d    (0xd00320f4 set 0x400);								  */
-	/* Loop timing	    (0xd003208c set 0x1000;								  */
+	/* Set power_reg0    (0xd0032004 set 0xfc00);recheck bits 7,2,3  */
+	/* Set register      (0x00032134 set 0x14);                      */
+	/* Set register      (0x00032034 set 0xc94c);correct amplitude   */
+	/* Set register 0x3d (0xd00320f4 set 0x400);                     */
+	/* Loop timing       (0xd003208c set 0x1000;                     */
 	status  = asicOntMiscRegWrite(mvAsicReg_PON_SERDES_INTERNAL_POWER_REG_0, 0xFC00, 0);
 	if (status != MV_OK)
 		return status;
 
 	mvOsDelay(20);
 
-	status  = asicOntMiscRegWrite(mvAsicReg_PON_SERDES_INTERNAL_REG_51, 0x104, 0);
+	status  = asicOntMiscRegWrite(mvAsicReg_PON_SERDES_INTERNAL_REG_4D, 0x14, 0);
+	if (status != MV_OK)
+		return status;
+
+	mvOsDelay(20);
+
+	status  = asicOntMiscRegWrite(mvAsicReg_PON_SERDES_INTERNAL_REG_D, 0xc94c, 0);
 	if (status != MV_OK)
 		return status;
 
-- 
1.7.5.4

